libs/pages/cluster/src/lib/ui/page-settings-general/page-settings-general.spec.tsx

```diff
@@ -0,0 +1,52 @@
+import { waitFor } from '@testing-library/react'
+import { render } from '__tests__/utils/setup-jest'
+import { wrapWithReactHookForm } from '__tests__/utils/wrap-with-react-hook-form'
+import PageSettingsGeneral, { PageSettingsGeneralProps } from './page-settings-general'
+
+describe('PageSettingsGeneral', () => {
+  const props: PageSettingsGeneralProps = {
+    onSubmit: jest.fn((e) => e.preventDefault()),
+    loading: false,
+  }
+
+  const defaultValues = {
+    name: 'hello-world',
+    description: 'desc',
+    production: true,
+  }
+
+  it('should render successfully', async () => {
+    const { baseElement } = render(wrapWithReactHookForm(<PageSettingsGeneral {...props} />))
+    expect(baseElement).toBeTruthy()
+  })
+
+  it('should render the form with fields', async () => {
+    const { getByDisplayValue } = render(
+      wrapWithReactHookForm(<PageSettingsGeneral {...props} />, {
+        defaultValues: defaultValues,
+      })
+    )
+
+    getByDisplayValue('hello-world')
+    getByDisplayValue('desc')
+    getByDisplayValue('true')
+  })
+
+  it('should submit the form', async () => {
+    const spy = jest.fn((e) => e.preventDefault())
```

Comment:
You don't need to create the spy here, you can just do expect(props.onSubmit).hasBeenCalled